camera_system/coordinate_getter.py

```diff
@@ -0,0 +1,81 @@
+"""座標取得GUIツール.
+
+画像を表示し、クリックした座標を保持する
+@author miyashita64 mutotaka0426
+@note 参考: https://techacademy.jp/magazine/51035
+"""
+
+import cv2
+from typing import List
+
+
+class CoordinateGetter:
+    """画像から座標を取得するクラス.
+
+    Attributes:
+        __block_point: ブロック置き場の座標リスト
+        __base_circle: ベースサークルの座標リスト
+    """
+
+    def __init__(self, file_name: str) -> None:
+        """カメラシステムのコンストラクタ.
+
+        Args:
+            file_name (str, optional): 入力画像ファイルのパス.
+        """
+        # メンバを初期化する.
+        self.__block_point = []
+        self.__base_circle = []
+
+        # GUIツールを設定する
+        self.__img = cv2.imread(file_name)
+        self.__window_name = "Coordinate Getter"
+        # 画像を表示する.
+        self.__showImage()
+
+    def __showImage(self) -> None:
+        """画像を表示する関数."""
+        cv2.imshow(self.__window_name, self.__img)
+        cv2.setMouseCallback(self.__window_name, self.__setCoordinate)
+        cv2.waitKey(0)

```

Comment:
上記のコンストラクタの代わりに、こちらの関数を呼び出すことで、座標入力画面と画像の表示を行うのはどうでしょうか？

また、このクラスを呼び出す際、画像は一旦ファイルに保存して受け渡すよりも、直接OpenCVのオブジェクトを受け渡したほうが簡潔に記述できると思います。

```suggestion
    def show_coordinate_input_window(self, img: cv2.Mat) -> None:
        """画像と座標を入力する画面を表示する関数.

        Args:
            img (cv2.Mat): 画像データ
        """
        self.__img = img
        cv2.imshow(self.__window_name, self.__img)
        cv2.setMouseCallback(self.__window_name, self.__setCoordinate)
        cv2.waitKey(0)
```

Reply:
画像データはコンストラクタで渡してしまっても良い気がしてきました。

Reply:
引数として渡すのは画像データよりもファイルパスの方がよくない？
呼び出し元でもopenCVを扱うより，CoordinateGetterの方でopenCV関連の処理を一任した方がよくない？

Reply:
一つの画像に対して一つのCoordinateGetterインスタンスであるべきな気がしてきた．

指摘通りコンストラクタとウィンドウ表示は分けるとして，コンストラクタにファイルパスは渡すのはどう？